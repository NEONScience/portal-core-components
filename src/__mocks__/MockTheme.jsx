import React from 'react';

import CssBaseline from '@mui/material/CssBaseline';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';

import Theme from '../lib_components/components/Theme/Theme';

const MockTheme = ({ children }) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={Theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default MockTheme;
