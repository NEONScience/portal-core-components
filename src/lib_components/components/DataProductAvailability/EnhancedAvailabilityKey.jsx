/* eslint-disable max-len, no-unused-vars, prefer-destructuring */

import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@mui/styles';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';

import CloseIcon from '@mui/icons-material/Close';
import HelpIcon from '@mui/icons-material/HelpOutline';

import { SVG, VALID_ENHANCED_STATUSES } from './AvailabilityUtils';
import { JsxCell } from './AvailabilitySvgComponents';

import Theme, { COLORS } from '../Theme/Theme';
import { resolveProps } from '../../util/defaultProps';

/**
   Setup: CSS classes
*/
const useStyles = makeStyles((theme) => ({
  keyContainer: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
  },
  keyElement: {
    cursor: 'help',
    margin: theme.spacing(0, 3, 0.25, 0),
  },
  keyElementText: {
    textAnchor: 'start',
    whiteSpace: 'pre',
    fontFamily: '"Cutive Mono","Lucida Console",Monaco,monospace',
    fontWeight: 400,
    fontSize: `${SVG.LABEL_FONT_SIZE}px`,
    fill: Theme.palette.grey[700],
  },
  h6Small: {
    fontSize: '0.95rem',
  },
}));

const ALL_SELECTED_TITLE = 'If the chart is presenting a roll-up (e.g. view by state) then all sites rolled into a given row are selected';
const SOME_SELECTED_TITLE = 'If the chart is presenting a roll-up (e.g. view by state) then one or more but not all of the sites rolled into a given row are selected';

const defaultProps = {
  selectionEnabled: false,
  rollUpPresent: false,
};

/**
   Main Function
*/
export default function EnhancedAvailabilityKey(inProps) {
  const props = resolveProps(defaultProps, inProps);
  const classes = useStyles(Theme);
  const [dialogOpen, setDialogOpen] = useState(false);
  const { selectionEnabled, rollUpPresent } = props;

  const statusSvgHeight = SVG.CELL_HEIGHT + 2;
  const labelLetterWidth = 8;
  const labelY = SVG.LABEL_FONT_SIZE - SVG.CELL_PADDING + 1;
  const statusLabelX = SVG.CELL_WIDTH + (2 * SVG.CELL_PADDING);

  // Let each element in the key be clickable to open the help dialog.
  // This is in addition to the help button, but we should still provide aria attributes.
  const dialogOpenerProps = {
    role: 'button',
    onClick: () => setDialogOpen(true),
  };

  const statusSvgs = {};
  Object.keys(VALID_ENHANCED_STATUSES).forEach((status) => {
    const label = VALID_ENHANCED_STATUSES[status].title;
    const description = VALID_ENHANCED_STATUSES[status].description;
    const statusSvgWidth = (label.length * labelLetterWidth) + statusLabelX;
    statusSvgs[status] = (
      <div className={classes.keyElement} title={description} {...dialogOpenerProps}>
        <svg width={statusSvgWidth} height={statusSvgHeight}>
          <JsxCell status={status} />
          <text className={classes.keyElementText} x={statusLabelX} y={labelY}>
            {label}
          </text>
        </svg>
      </div>
    );
  });

  const renderSelectionElement = (variant, inDialog = false) => {
    if (!['all', 'some'].includes(variant)) { return null; }
    const selectionSvgHeight = SVG.CELL_HEIGHT + 2;
    const label = variant === 'all' ? 'All sites selected' : 'Some sites selected';
    const fill = variant === 'all' ? Theme.palette.primary.main : 'url(#partialSelectionPattern)';
    const description = variant === 'all' ? ALL_SELECTED_TITLE : SOME_SELECTED_TITLE;
    const selectionWidth = 45;
    const selectionLabelX = selectionWidth + (3 * SVG.CELL_PADDING);
    const selectionSvgWidth = (label.length * labelLetterWidth) + selectionLabelX;
    const handleAttribs = {
      width: SVG.DATE_RANGE_HANDLE_WIDTH,
      height: SVG.CELL_HEIGHT,
      fill: COLORS.LIGHT_BLUE[300],
      stroke: Theme.palette.primary.main,
      strokeWidth: 1.5,
    };
    const graphic = (
      <>
        <rect x={0.5} y={1.5} width={selectionWidth} height={SVG.CELL_HEIGHT - 2} fill={fill} />
        <rect x={0.5} y={0.5} {...handleAttribs} />
        <rect x={selectionWidth - SVG.DATE_RANGE_HANDLE_WIDTH} y={0.5} {...handleAttribs} />
      </>
    );
    return inDialog ? (
      <div style={{ marginBottom: Theme.spacing(2) }}>
        <div style={{ display: 'flex', alignItems: 'center' }}>
          <svg
            width={Math.ceil(selectionWidth * 1.25)}
            height={Math.ceil(selectionSvgHeight * 1.25)}
            viewBox={`0 0 ${selectionWidth} ${selectionSvgHeight}`}
            style={{ marginRight: Theme.spacing(1) }}
          >
            {graphic}
          </svg>
          <Typography variant="subtitle2" style={{ fontSize: '1.05rem' }}>{label}</Typography>
        </div>
        <Typography variant="body2">{description}</Typography>
      </div>
    ) : (
      <div className={classes.keyElement} title={description} {...dialogOpenerProps}>
        <svg width={selectionSvgWidth} height={statusSvgHeight}>
          {graphic}
          <text className={classes.keyElementText} x={selectionLabelX} y={labelY}>
            {label}
          </text>
        </svg>
      </div>
    );
  };

  const renderDialog = () => {
    const renderStatus = (status) => {
      const { title, description } = VALID_ENHANCED_STATUSES[status];
      return (
        <div style={{ marginBottom: Theme.spacing(2.5) }}>
          <div style={{ display: 'flex', alignItems: 'center' }}>
            <svg
              width={Math.ceil(SVG.CELL_WIDTH * 1.25)}
              height={Math.ceil(SVG.CELL_HEIGHT * 1.25)}
              viewBox={`0 0 ${SVG.CELL_WIDTH} ${SVG.CELL_HEIGHT}`}
              style={{ marginRight: Theme.spacing(1) }}
            >
              <JsxCell status={status} />
            </svg>
            <Typography variant="subtitle2" style={{ fontSize: '0.95rem', marginTop: '2px' }}>
              {title}
            </Typography>
          </div>
          <Typography variant="body2">{description}</Typography>
        </div>
      );
    };
    return (
      <Dialog
        open={dialogOpen}
        maxWidth="md"
        onClose={() => setDialogOpen(false)}
        aria-labelledby="availability-key-dialog-title"
      >
        <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
          <DialogTitle id="availability-key-dialog-title">
            <span style={{ fontSize: '1.4rem', fontWeight: '600' }}>
              Data Availability Chart Key
            </span>
          </DialogTitle>
          <IconButton
            title="Close"
            aria-label="Close"
            onClick={() => setDialogOpen(false)}
            style={{ marginRight: Theme.spacing(1) }}
            size="large"
          >
            <CloseIcon fontSize="inherit" />
          </IconButton>
        </div>
        <DialogContent style={{ marginBottom: Theme.spacing(2) }}>
          <div style={{ display: 'flex', alignItems: 'flex-start' }}>
            <div style={{ flex: '50%', marginRight: Theme.spacing(2) }}>
              {renderStatus('available')}
              {renderStatus('being processed')}
              {renderStatus('expected')}
              {renderStatus('not expected')}
            </div>
            <div style={{ flex: '50%' }}>
              {renderStatus('not available')}
              {renderStatus('not collected')}
              {renderStatus('delayed')}
              {renderStatus('tentative')}
            </div>
          </div>
          <div style={{ display: 'flex', alignItems: 'flex-start' }}>
            <div style={{ flex: '50%', marginRight: Theme.spacing(2) }}>
              {renderStatus('mixed some availability')}
            </div>
            <div style={{ flex: '50%' }}>
              {renderStatus('mixed no availability')}
            </div>
          </div>
          <div style={{ display: 'flex', alignItems: 'flex-start' }}>
            <div style={{ flex: '50%', marginRight: Theme.spacing(2) }}>
              {renderSelectionElement('all', true)}
            </div>
            <div style={{ flex: '50%' }}>
              {renderSelectionElement('some', true)}
            </div>
          </div>
        </DialogContent>
      </Dialog>
    );
  };

  return (
    <>
      <div className={classes.keyContainer}>
        <div>
          <Typography
            variant="h6"
            className={classes.h6Small}
            style={{ margin: Theme.spacing(-0.75, 3, 0.5, 0) }}
          >
            Key:
          </Typography>
          <IconButton
            size="small"
            color="primary"
            title="Help - Data Availability Chart Key"
            aria-label="Help - Data Availability Chart Key"
            onClick={() => setDialogOpen(true)}
            style={{ marginLeft: '-3px' }}
          >
            <HelpIcon fontSize="inherit" />
          </IconButton>
        </div>
        <div>
          {statusSvgs.available}
          {statusSvgs['being processed']}
          {statusSvgs.expected}
          {statusSvgs['not expected']}
        </div>
        <div>
          {statusSvgs['not available']}
          {statusSvgs['not collected']}
          {statusSvgs.delayed}
          {statusSvgs.tentative}
        </div>
        {!(selectionEnabled || rollUpPresent) ? null : (
          <div>
            {!rollUpPresent ? null : (
              <>
                {statusSvgs['mixed some availability']}
                {statusSvgs['mixed no availability']}
              </>
            )}
            {!selectionEnabled ? null : (
              <>
                {renderSelectionElement('all')}
                {renderSelectionElement('some')}
              </>
            )}
          </div>
        )}
      </div>
      {renderDialog()}
    </>
  );
}

EnhancedAvailabilityKey.propTypes = {
  selectionEnabled: PropTypes.bool,
  rollUpPresent: PropTypes.bool,
};
