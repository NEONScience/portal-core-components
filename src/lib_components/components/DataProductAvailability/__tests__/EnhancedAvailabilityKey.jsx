import React from 'react';
import renderer from 'react-test-renderer';

import MockTheme from '../../../../__mocks__/MockTheme';
import EnhancedAvailabilityKey from '../EnhancedAvailabilityKey';

describe('DataProductAvailability - EnhancedAvailabilityKey', () => {
  test('renders correctly when selection is not enabled (default)', () => {
    const tree = renderer
      .create(<MockTheme><EnhancedAvailabilityKey /></MockTheme>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders correctly when selection is enabled', () => {
    const tree = renderer
      .create(<MockTheme><EnhancedAvailabilityKey selectionEnabled /></MockTheme>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
