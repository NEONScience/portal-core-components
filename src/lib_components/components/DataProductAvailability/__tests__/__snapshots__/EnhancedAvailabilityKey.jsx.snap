// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataProductAvailability - EnhancedAvailabilityKey renders correctly when selection is enabled 1`] = `
<div
  className="makeStyles-keyContainer-5"
>
  <div>
    <h6
      className="MuiTypography-root MuiTypography-h6 makeStyles-h6Small-8 css-134ad2t-MuiTypography-root"
      style={
        {
          "margin": "-6px 24px 4px 0px",
        }
      }
    >
      Key:
    </h6>
    <button
      aria-label="Help - Data Availability Chart Key"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-8spxgg-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "marginLeft": "-3px",
        }
      }
      tabIndex={0}
      title="Help - Data Availability Chart Key"
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data have been published and released. Data are available for download"
    >
      <svg
        height={14}
        width={151}
      >
        <rect
          fill="#002c77"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Release Available
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data have been collected and will be published after processing has completed"
    >
      <svg
        height={14}
        width={135}
      >
        <rect
          fill="url(#beingProcessedPattern)"
          height="10.5px"
          rx="1px"
          stroke="#002c77"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Being Processed
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data collection is scheduled to happen"
    >
      <svg
        height={14}
        width={79}
      >
        <rect
          fill="#ffffff"
          height="10.5px"
          rx="1px"
          stroke="#002c77"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Expected
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="No data collection is scheduled"
    >
      <svg
        height={14}
        width={111}
      >
        <rect
          fill="#ffffff"
          height="10.5px"
          rx="1px"
          stroke="#d7d9d9"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Not Expected
        </text>
      </svg>
    </div>
  </div>
  <div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data was neither expected nor collected"
    >
      <svg
        height={14}
        width={119}
      >
        <rect
          fill="#d7d9d9"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Not Available
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data collection was expected but collection could not take place; no data will be published"
    >
      <svg
        height={14}
        width={119}
      >
        <rect
          fill="#f8bb32"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Not Collected
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data should be available for download but something has prevented publication"
    >
      <svg
        height={14}
        width={71}
      >
        <rect
          fill="url(#delayedPattern)"
          height="10.5px"
          rx="1px"
          stroke="#f8bb32"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Delayed
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="Data collection is desired but not expected until other conditions are met"
    >
      <svg
        height={14}
        width={87}
      >
        <rect
          fill="#ffffff"
          height="10.5px"
          rx="1px"
          stroke="#b39076"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={15}
          y={9}
        >
          Potential
        </text>
      </svg>
    </div>
  </div>
  <div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="If the chart is presenting a roll-up (e.g. view by state) then all sites rolled into a given row are selected"
    >
      <svg
        height={14}
        width={201}
      >
        <rect
          fill="#0073cf"
          height={10}
          width={45}
          x={0.5}
          y={1.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={0.5}
          y={0.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={41}
          y={0.5}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={57}
          y={9}
        >
          All sites selected
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-6"
      onClick={[Function]}
      role="button"
      title="If the chart is presenting a roll-up (e.g. view by state) then one or more but not all of the sites rolled into a given row are selected"
    >
      <svg
        height={14}
        width={209}
      >
        <rect
          fill="url(#partialSelectionPattern)"
          height={10}
          width={45}
          x={0.5}
          y={1.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={0.5}
          y={0.5}
        />
        <rect
          fill="#5ca6e3"
          height={12}
          stroke="#0073cf"
          strokeWidth={1.5}
          width={4}
          x={41}
          y={0.5}
        />
        <text
          className="makeStyles-keyElementText-7"
          x={57}
          y={9}
        >
          Some sites selected
        </text>
      </svg>
    </div>
  </div>
</div>
`;

exports[`DataProductAvailability - EnhancedAvailabilityKey renders correctly when selection is not enabled (default) 1`] = `
<div
  className="makeStyles-keyContainer-1"
>
  <div>
    <h6
      className="MuiTypography-root MuiTypography-h6 makeStyles-h6Small-4 css-134ad2t-MuiTypography-root"
      style={
        {
          "margin": "-6px 24px 4px 0px",
        }
      }
    >
      Key:
    </h6>
    <button
      aria-label="Help - Data Availability Chart Key"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-8spxgg-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "marginLeft": "-3px",
        }
      }
      tabIndex={0}
      title="Help - Data Availability Chart Key"
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </button>
  </div>
  <div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data have been published and released. Data are available for download"
    >
      <svg
        height={14}
        width={151}
      >
        <rect
          fill="#002c77"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Release Available
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data have been collected and will be published after processing has completed"
    >
      <svg
        height={14}
        width={135}
      >
        <rect
          fill="url(#beingProcessedPattern)"
          height="10.5px"
          rx="1px"
          stroke="#002c77"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Being Processed
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data collection is scheduled to happen"
    >
      <svg
        height={14}
        width={79}
      >
        <rect
          fill="#ffffff"
          height="10.5px"
          rx="1px"
          stroke="#002c77"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Expected
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="No data collection is scheduled"
    >
      <svg
        height={14}
        width={111}
      >
        <rect
          fill="#ffffff"
          height="10.5px"
          rx="1px"
          stroke="#d7d9d9"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Not Expected
        </text>
      </svg>
    </div>
  </div>
  <div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data was neither expected nor collected"
    >
      <svg
        height={14}
        width={119}
      >
        <rect
          fill="#d7d9d9"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Not Available
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data collection was expected but collection could not take place; no data will be published"
    >
      <svg
        height={14}
        width={119}
      >
        <rect
          fill="#f8bb32"
          height="12px"
          rx="2px"
          stroke={null}
          strokeWidth={null}
          width="7px"
          x={0}
          y={0}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Not Collected
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data should be available for download but something has prevented publication"
    >
      <svg
        height={14}
        width={71}
      >
        <rect
          fill="url(#delayedPattern)"
          height="10.5px"
          rx="1px"
          stroke="#f8bb32"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Delayed
        </text>
      </svg>
    </div>
    <div
      className="makeStyles-keyElement-2"
      onClick={[Function]}
      role="button"
      title="Data collection is desired but not expected until other conditions are met"
    >
      <svg
        height={14}
        width={87}
      >
        <rect
          fill="#ffffff"
          height="10.5px"
          rx="1px"
          stroke="#b39076"
          strokeWidth="1.5px"
          width="5.5px"
          x={0.75}
          y={0.75}
        />
        <text
          className="makeStyles-keyElementText-3"
          x={15}
          y={9}
        >
          Potential
        </text>
      </svg>
    </div>
  </div>
</div>
`;
